### ISSUE TITLE
Torrents are downloading but file percentage not increasing.

### ISSUE THREAD

**Issue by swoondrones** [transmission/transmission#7079]
### What is the issue?

Has anyone else come across this? I download files and I find that I'm using a lot of data downloading a torrent but the percentage of the file doesn't increase. It stays the same. Anyone seen that before? I've had a few torrent do that. Can someone please fix this? That would be so cool!

### Which application of Transmission?

macOS app

### Which version of Transmission?

406

**Comment by killemov** [comment]
To be clear, this happens sporadically? Does it happen with torrents that have some files deselected?

**Comment by swoondrones** [comment]
> To be clear, this happens sporadically? Does it happen with torrents that have some files deselected?

Hey, no. I have everything selected. What I see is dats ticking over, but the percentage stays the same. I wonder where the data is going as it is not in the file the torrent is downloading. I have had 2 files stop just shy of 100% and another file stop at 90%, even though they are downloading data from somewhere. It is so weird.

**Comment by tearfur** [comment]
Similar problems are being reported from a long while ago. We know that one of the causes is that Transmission is not handling a full disk properly, but people with plenty of disk space have seen this too, and these people are mainly on macOS. Sadly, somehow nobody with developer skills have managed to reproduce and reported their findings...

- #6796
- #6691
- #6579
- #6011
- #5747

**Comment by killemov** [comment]
Yeah, at least for preventing a full disk I again suggest implementing a low water mark system. See #1835.

**Comment by wally-crunk** [comment]
@swoondrones When this happens, have you gone to Window > Message Log (⌘3) > Debug and reviewed for any messages?  Each torrent being downloaded should generate many (perhaps dozens of) messages, like 
> ```
> 2024-09-09 19:01:27 +0000 open-files.cc:47 [Debug] open-files.cc:47: Fast preallocation failed: No such file or directory (2)
> 2024-09-09 19:01:27 +0000 open-files.cc:222 [Debug] open-files.cc:222: Preallocated file '/Users/foo/Downloads/ubuntu-24.04.1-desktop-amd64.iso.part' (sparse, size: 12345)
> ``` 

Can you show your messages?  Replacing the filenames with foo.bar or xxx may be appropriate.

**Comment by axi** [comment]
When you let that bug live...
![image](https://github.com/user-attachments/assets/97eb83ac-071d-468f-842f-81099212ca8a)
Restarting the daemon (linux) did help
